### DISCUSSION TITLE
Use Python(PyQt) to Measuring Sound with Microphones

### DISCUSSION THREAD

**Discussion by chia-kai-chang** [1101-GS4719-C/Python-Community-GS4719-C#1380] [General]
這是一則轉貼文章。

影片展示了，在Windows / Linux OS中使用Python完成Microphones聲音資料擷取與Spectrogram(時頻譜)圖繪製。

應該是使用SciPy套件做傅立葉轉換，然後用PyQt即時顯示在視窗介面上。

原文章作者應認為Python的應用於工業自動控制的困難點在於「難以將資料圖表快速即時視覺化呈現於視窗介面」，所以標題下了PyQt。但其實要能分析聲音時域資訊成頻域，SciPy也是不可少的套件。

youtube影片：https://youtu.be/iHOifA6WT1k

也可以在這邊看：https://labview360.org/t/topic/32652

**Comment by quan0715** [comment]
教授好棒，PyQT 框架真的可以幫助同學們快速開發DEMO介面，但唯獨GUI介面都偏難看，且有時後手拉GUI對沒有 Layout 觀念的同學有時候也是種折磨，不過至少比手刻好了。 如果同學想要好好美化你的app的話可以去研究一下QSS，他會打破你的醜醜GUI的既定印象
